servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/UdpReporter.java

```diff
@@ -0,0 +1,252 @@
+                builder.ioExecutor != null ? builder.ioExecutor : globalExecutionContext().ioExecutor()
+        ).eventLoopGroup();
+        try {
+            final Bootstrap bootstrap = buildBootstrap(group, builder.codec, builder.collectorAddress,
+                    builder.loggerName);
+            channel = bootstrap.bind(0).sync().channel();
+        } catch (InterruptedException e) {
+            currentThread().interrupt(); // Reset the interrupted flag.
+            throw new IllegalStateException("Failed to create UDP client");
+        } catch (Exception e) {
+            logger.warn("Failed to create UDP client", e);
+            throw e;
+        }
+        Executor executor = builder.executor != null ? builder.executor :
+                globalExecutionContext().executor();
+        closeable = new NettyChannelListenableAsyncCloseable(channel, executor);
+    }
+
+    /**
+     * The serialization format for the zipkin write format data.
+     */
+    public enum Codec {
+        JSON_V1(SpanBytesEncoder.JSON_V1), JSON_V2(SpanBytesEncoder.JSON_V2),
+        THRIFT(SpanBytesEncoder.THRIFT), PROTO3(SpanBytesEncoder.PROTO3);
+
+        final SpanBytesEncoder encoder;
+
+        Codec(SpanBytesEncoder encoder) {
+            this.encoder = encoder;
+        }
+    }
+
+    /**
+     * A builder to create a new {@link UdpReporter}.
+     */
+    public static final class Builder {
+        private final SocketAddress collectorAddress;
+        private Codec codec = Codec.JSON_V2;
+        @Nullable
+        private IoExecutor ioExecutor;
+        @Nullable
+        private Executor executor;
+        @Nullable
+        private String loggerName;
+
+        /**
+         * Create a new {@link UdpReporter.Builder} for a given collectorAddress.
+         *
+         * @param collectorAddress the collector SocketAddress
+         */
+        public Builder(SocketAddress collectorAddress) {
+            this.collectorAddress = collectorAddress;
+        }
+
+        /**
+         * Sets the {@link UdpReporter.Codec} to encode the Spans with.
+         *
+         * @param codec the codec to use for this span.
+         * @return {@code this}
+         */
+        public Builder codec(Codec codec) {
+            this.codec = requireNonNull(codec);
+            return this;
+        }
+
+        /**
+         * Sets an {@link IoExecutor} to use for writing to the datagram channel. Defaults to the
+         * {@link GlobalExecutionContext} {@link IoExecutor}.
+         *
+         * @param ioExecutor IoExecutor to use to write with.
+         * @return {@code this}
+         */
+        public Builder ioExecutor(IoExecutor ioExecutor) {
+            this.ioExecutor = requireNonNull(ioExecutor);
+            return this;
+        }
+
+        /**
+         * Sets an {@link Executor} to use for offloading various tasks. Defaults to the {@link GlobalExecutionContext}
+         * {@link Executor}.
```

Comment:
```suggestion
         * Sets an {@link Executor} to use when required. 
```

Avoid stating defaults in documentation.

Reply:
Why?

Reply:
Stating defaults in docs means that we have less flexibility to change them in the future. If a user cares about a default value (more than just knowledge), it is a better option to explicitly use that default value so that if the defaults change they are not surprised.